homeassistant/components/keba/binary_sensor.py

```diff
@@ -0,0 +1,147 @@
+"""Support for KEBA charging station binary sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS = {
+    'Online': ['Wallbox', 'connectifity'],
+    'Plug': ['Plug', 'plug'],
+    'State': ['Charging state', 'power'],
+    'Tmo FS': ['Failsafe Mode', 'safety'],
+}
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station binary sensors")
+    sensors = []
+    keba = hass.data[DOMAIN]
+
+    for key, value in sorted(SENSORS.items()):
+        sensors.append(KebaBinarrySensor(key, keba, value[0], value[1]))
+
+    async_add_entities(sensors)
+
+
+class KebaBinarrySensor(BinarySensorDevice):
+    """Representation of a binary sensor of a KEBA charging station."""
+
+    def __init__(self, key, keba, sensor_name, device_class):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._name = sensor_name
+        self._device_class = device_class
+        self._state = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Return False.
+
+        Data update is triggered from Keba entitiy.
+        """
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        return "keba_" + self._name
```

Comment:
Are names unique even when using multiple charging stations?

Reply:
Name is not allowed to be used for unique IDs. platform name is not necessary to be included in unique id.

Reply:
I added a device name for the KebaHandler object, which includes the host (ip address). This device_name is now used as part of the unique_id (device_name + sensor_name).

Reply:
Host/IP address is not to be used for unique ids as it may change, see https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements for ideas of good candidates for this (something that is bound to the HW).

Reply:
Reading the documentation would have helped, I will use the serial number then ;)